src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/helpers/PickRandomHelper.java

```diff
@@ -40,20 +40,8 @@ public Object apply(Object context, Options options) throws IOException {
 
     Integer count = (Integer) options.hash.get("count");
     if (count != null && count > 0) {
-      final List<Object> outputValueList = new ArrayList<>(count);
-      final Set<Integer> usedIndexes = new HashSet<>();
-
-      for (int i = 0; i < count && usedIndexes.size() < candidateValueList.size(); i++) {
-        int index = -1;
-        while (index == -1 || usedIndexes.contains(index)) {
-          index = ThreadLocalRandom.current().nextInt(candidateValueList.size());
-        }
-
-        outputValueList.add(candidateValueList.get(index));
-        usedIndexes.add(index);
-      }
-
-      return outputValueList;
+      Collections.shuffle(candidateValueList);
```

Comment:
Didn't do it this way due to performance. Shuffling a list of 100s just to pick 10 at random seems inefficient.

Reply:
good point. ive pushed another commit that only shuffles the list up to the desired length. also added a check to make sure the helper is returning a list of the correct length, because i realised we werent actually doing that. will revert again if you still dont think this is more efficient.